applications/CompressiblePotentialFlowApplication/custom_processes/define_3d_wake_process.cpp

```diff
@@ -0,0 +1,878 @@
+// by avaraging the local wake normals of the surrounding conditions
+void Define3DWakeProcess::ComputeAndSaveLocalWakeNormal() const
+{
+    // If mrTrailingEdgeModelPart doesn't have conditions use the global wake normal
+    if(mrTrailingEdgeModelPart.NumberOfConditions() < 1){
+        // Use global wake normal
+        for (auto& r_node : mrTrailingEdgeModelPart.Nodes()){
+            r_node.SetValue(WAKE_NORMAL, mWakeNormal);
+        }
+    }
+    else{
+        for (auto& r_cond : mrTrailingEdgeModelPart.Conditions()){
+            auto& r_geometry = r_cond.GetGeometry();
+            const auto& coordinates1 = r_geometry[0].Coordinates();
+            const auto& coordinates2 = r_geometry[1].Coordinates();
+
+            const auto& local_span_direction = coordinates2 - coordinates1;
+            array_1d<double, 3> local_wake_normal = ZeroVector(3);
+            MathUtils<double>::CrossProduct(local_wake_normal, mWakeDirection, local_span_direction);
+
+            const double projection = inner_prod(local_wake_normal, mWakeNormal);
+
+            if(projection < 0.0){
+                local_wake_normal *= -1.0;
+            }
+
+            for (unsigned int i = 0; i < r_geometry.size(); i++){
+                r_geometry[i].GetValue(WAKE_NORMAL) += local_wake_normal;
+            }
+        }
+
+        for (auto& r_node: mrTrailingEdgeModelPart.Nodes()){
+            auto& local_wake_normal = r_node.GetValue(WAKE_NORMAL);
+            const double norm = MathUtils<double>::Norm3(local_wake_normal);
+            local_wake_normal /= norm;
+        }
+    }
+}
+
+// This function creates the wake surface automatically by shedding it from the
+// trailing edge in the dicretion of the free stream velocity (mWakeDirection).
+// The user can decide how much distance is to be shedded and the element size
+// of the wake surface in the wake direction. Note that the element size in span
+// direction is predetermined by the size of the conditions constituting the
+// trailing edge.
+void Define3DWakeProcess::ShedWakeSurfaceFromTheTrailingEdge()
+{
+    const Properties::Pointer pElemProp = mrStlWakeModelPart.pGetProperties(0);
+    const double number_of_elements = mSheddedWakeDistance / mSheddedWakeElementSize;
+    const unsigned int number_of_elements_in_wake_direction = int(number_of_elements);
+
+    IndexType node_index = 0;
+    IndexType element_index = 0;
+
+    array_1d<double,3> coordinates1 = ZeroVector(3);
+    array_1d<double,3> coordinates2 = ZeroVector(3);
+    array_1d<double,3> coordinates3 = ZeroVector(3);
+    array_1d<double,3> coordinates4 = ZeroVector(3);
+
+    for (auto& r_cond : mrTrailingEdgeModelPart.Conditions()) {
+        auto& r_geometry = r_cond.GetGeometry();
+        coordinates1 = r_geometry[0].Coordinates();
+        coordinates2 = r_geometry[1].Coordinates();
+
+        if(r_geometry[0].GetValue(WING_TIP) != 0){
+            DecreaseWakeWidthAtTheWingTips(coordinates1, coordinates2);
```

Comment:
I'm not sure what you refer to as bad cuts. The intention was not really to avoid bad cuts, but to avoid cutting elements on the wake sides at all. At the beginning I thought this was not a problem, but I ran into some issues when putting the wake exactly at the wing tip, I can show you later what I mean.

Reply:
I think I understand what you mean. If the wake stl is a bit further than the wing tip the elements that are "outside" of the span are also marked as wake no? I was referring to this yes!

Reply:
Yes exactly